core/genesis.go

```diff
@@ -161,7 +161,7 @@ func (ga *GenesisAlloc) flush(db ethdb.Database, triedb *trie.Database, blockhas
 	}
 
 	for addr, account := range *ga {
-		statedb.AddBalance(addr, account.Balance)
+		statedb.SetBalance(addr, account.Balance)
```

Comment:
This might seem weird, but clearly both options are equivalent. When you create the verkle genesis, adding the balance and setting the balance should be the same since the previous balance is 0.

This is required for the situation described in the comment below in this file. 

When the genesis block is regenerated and "re-saved", the stated will use an already existing VKT (since for VKT, we only have a single tree on disk).

But... if creating the genesis uses `AddBalance` that is bad! We'll add the genesis balance for accounts twice, messing up with the expected hash root. That's why `SetBalance(...)` is safer. 

The reality is that this `AddBalance`->`SetBalance` shouldn't be needed if whenever you open a `statedb.New(...)` it would respect whatever `root` you asked for (i.e: versioned VKTs). Since now this is ignored, we get into this situation where this matters.

Reply:
ok I don't like it but in the spirit of saving time, I will merge this and make a note that we need to revisit this after the rebase.